astropy/io/ascii/daophot.py

```diff
@@ -12,15 +12,84 @@
 
 import re
 import numpy as np
+import itertools as itt
+from collections import defaultdict
+
 from . import core
-from . import basic
 from . import fixedwidth
 from ...utils import OrderedDict
+from ...extern.six.moves import zip, map
+from .misc import first_true_index, first_false_index, groupmore
 
 
 class DaophotHeader(core.BaseHeader):
-    comment = r'\s*#K'
-    aperture_values = ''
+    """Read the header from a file produced by the IRAF DAOphot routine."""
+
+    def __init__(self):
+        core.BaseHeader.__init__(self)
+        self.comment = r'\s*#K'
+
+        # Regex for extracting the format strings
+        self.re_format = re.compile('%-?(\d+)\.?\d?[sdfg]')
+        self.re_header_keyword = re.compile(r'[#]K'
+                                            r'\s+ (?P<name> \w+)'
+                                            r'\s* = (?P<stuff> .+) $',
+                                            re.VERBOSE)
+
+        self.aperture_values = ()
+
+    def parse_col_defs(self, grouped_lines_dict):
+        """
+        Parse a series of column defintion lines like below.  There may be several
+        such blocks in a single file (where continuation characters have already been
+        stripped).
+        #N ID    XCENTER   YCENTER   MAG         MERR          MSKY           NITER
+        #U ##    pixels    pixels    magnitudes  magnitudes    counts         ##
+        #F %-9d  %-10.3f   %-10.3f   %-12.3f     %-14.3f       %-15.7g        %-6d
+        """
+        line_ids = ('#N', '#U', '#F')
+        coldef_dict = defaultdict(list)
+
+        # Function to strip identifier lines
+        stripper = lambda s: s[2:].strip(' \\')
```

Comment:
My linter complains when I assigned a lambda expression and says "if you assign it, use def."
Not sure if that's part of PEP8 though.